system/modules/system-partition.nix

```diff
@@ -53,5 +53,15 @@ in
       ln -s mnt/system/init init
       cd /
     '';
+
+    # Replace /dev/root by /dev/disk/by-label/system.x in /etc/fstab
+    boot.postBootCommands = with pkgs; ''
+      SYSTEM_DEVICE=$(readlink /dev/root)
+      mv /etc/fstab /etc/fstab.old
+      cp -L /etc/fstab.old /etc/fstab
```

Comment:
Two questions:

1. Why `mv` and `cp` back, instead of just copying?
2. Is there any use in keeping the `.old` copy around?

Reply:
`/etc/fstab` is a symlink to `/etc/static/fstab`, wich is itself a symlink to `/nix/store/hash-etc-fstab`. We’re first moving the `/etc/fstab` link so that we can copy its content to `/etc/fstab`.

Another solution: get fstab nix store path with `read -f /etc/fstab`, remove `/etc/fstab`, cp nix store fstab to `/etc/fstab`.

We can remove the`.old` file, though it has no harm as it is.